WildMeOrg/codex-frontend
# Dex 761 social group site settings

# Pull request checklist:
 - [x] Features and bugfixes should be PRed into the `develop` branch, **not** `main`
 - [x] All text is internationalized
     - There were some exhaustive-deps and no-unstable-nested-components warnings/errors in DefaultFieldTable.jsx, but the latter were all customBodyRender errors (and I didn't add any more linter errors than already existed on this file). I did a quick spot check, and it doesn't look like we have implemented a fix to this issue yet, but please correct me if I'm wrong.
 - [x] New features support all states (loading, error, etc) 
     - N/A

Thanks for keeping it clean! Feel free to merge your own pull request after it has been approved - just use the "squash & merge" option.

# QA Notes 
Users should be able to add and remove new social group roles. 
Users should not be able to save if any roles are empty strings or if there are any duplicates.
Users should be able to designate whether or not a particular role is usable by multiple individuals within a social group.

<img width="1840" alt="Screen Shot 2022-07-07 at 4 14 48 PM" src="https://user-images.githubusercontent.com/2775448/177886828-08d0438a-4230-48a2-9dd8-f991d589966a.png">

<img width="1840" alt="Screen Shot 2022-07-07 at 4 12 22 PM" src="https://user-images.githubusercontent.com/2775448/177886900-f84be115-fc29-484c-bf72-c39bee208886.png">

<img width="1840" alt="Screen Shot 2022-07-07 at 4 11 58 PM" src="https://user-images.githubusercontent.com/2775448/177886936-7805b2b9-b5e6-458a-8c1a-a174e402e546.png">

<img width="1840" alt="Screen Shot 2022-07-07 at 4 11 50 PM" src="https://user-images.githubusercontent.com/2775448/177886979-a0a8face-7359-49d8-8ac5-a0c2a1bc81b1.png">
